src/main/java/net/william278/velocitab/packet/ScoreboardManager.java

```diff
@@ -122,50 +120,56 @@ public void unVanishPlayer(@NotNull Player player) {
-
-        siblings.forEach(server -> server.getPlayersConnected().stream().filter(p -> p != player).forEach(connected -> dispatchPacket(packet, connected)));
+        final Optional<TabPlayer.Nametag> cachedTag = Optional.ofNullable(nameTags.getOrDefault(teamName, null));
+        cachedTag.ifPresent(nametag -> {
+            final UpdateTeamsPacket packet = UpdateTeamsPacket.create(
+                    plugin, createdTeams.get(player.getUniqueId()),
+                    "", nametag, player.getUsername()
+            );
+            siblings.forEach(server -> server.getPlayersConnected().stream().filter(p -> p != player)
+                    .forEach(connected -> dispatchPacket(packet, connected)));
+        });
     }
 
     public void updateRole(@NotNull Player player, @NotNull String role) {
         if (!player.isActive()) {
             plugin.getTabList().removeOfflinePlayer(player);
             return;
         }
-
         final String name = player.getUsername();
         final TabPlayer tabPlayer = plugin.getTabList().getTabPlayer(player).orElseThrow();
 
-        tabPlayer.getNametag(plugin).thenAccept(nametag -> {
-            final String[] split = nametag.split(Pattern.quote(player.getUsername()), 2);
+        tabPlayer.getNametag(plugin).thenAccept(tag -> {
+            final String[] split = tag.split(Pattern.quote(player.getUsername()), 2);
             final String prefix = split[0];
             final String suffix = split.length > 1 ? split[1] : "";
 
+            final TabPlayer.Nametag newTag = new TabPlayer.Nametag(prefix, suffix);
             if (!createdTeams.getOrDefault(player.getUniqueId(), "").equals(role)) {
                 if (createdTeams.containsKey(player.getUniqueId())) {
-                    dispatchGroupPacket(UpdateTeamsPacket.removeTeam(plugin, createdTeams.get(player.getUniqueId())), player);
+                    dispatchGroupPacket(
+                            UpdateTeamsPacket.removeTeam(plugin, createdTeams.get(player.getUniqueId())),
+                            player
+                    );
                 }
 
                 createdTeams.put(player.getUniqueId(), role);
-                this.nameTags.put(role, prefix + NAMETAG_DELIMITER + suffix);
-                dispatchGroupPacket(UpdateTeamsPacket.create(plugin, role, "", prefix, suffix, name), player);
-            } else if (!this.nameTags.getOrDefault(role, "").equals(prefix + NAMETAG_DELIMITER + suffix)) {
-                this.nameTags.put(role, prefix + NAMETAG_DELIMITER + suffix);
-                dispatchGroupPacket(UpdateTeamsPacket.changeNameTag(plugin, role, prefix, suffix), player);
+                this.nameTags.put(role, newTag);
+                dispatchGroupPacket(
+                        UpdateTeamsPacket.create(plugin, role, "", newTag, name),
+                        player
+                );
+            } else if (!this.nameTags.containsKey(role) && this.nameTags.get(role).equals(newTag)) {
```

Comment:
This if block is unreachable

Reply:
Could you elaborate?

Reply:
I mean the condition is alway false. If nameTags doesn't contain the key "role" the nameTags.get(role) will return null.

Reply:
This would thrown a NullPointerException

Reply:
I think the change I pushed should fix this? It was just a case of an erroneous `!` operator